### PR TITLE
@anandaroop: Places (v2)

### LINKED ISSUE #8: Admin panel + API
So my current thinking about this repo/module is that: it's hard to manage, and it's inefficient. We just arbitrarily execute some code when it's required and load all the data into memory, not great.

On Friday I slapped together a admin panel app for this and figured that we can just serve the data through a little API and cache it with a long expiry.

Before I get too deep into that though I was considering an alternative to running an actual server:

Use http://geojson.io/ to edit place geometry. We can authenticate with Github through that and store the geojson files in this repo. [Github has nice support for geojson.](https://github.com/blog/1528-there-s-a-map-for-that).

We can then just write a simple build process that deploys the JSON to S3/Cloudfront and just treat that as our 'API'

### PR DESCRIPTION
**Pull request by dzucconi**
This is an attempt at a small admin panel / service to provide locations data for querying partners.

It's intended to expose an API that will replace https://github.com/artsy/places + should close https://github.com/artsy/places/issues/8

In the end it seemed to make the most sense to me to write a small app to deal with this rather than concocting some kind of workflow using Github.

This app would also take over one-off weird geo-related endpoints in Force, such as https://github.com/artsy/force/blob/2ff2ea49315e6132eb73c0aae2b9f993e0d099ea/apps/geo/routes.coffee

---

The actual code here is pretty prototype-y (in particular the client-side code). I just wanted to get something workable together. So I wouldn't recommend actually forking and trying to run this, yet.

---

Ultimately we'll be replacing "Cities" which are points that are queried with a hardcoded radius, with "Places". Places in this case are just polygons of coordinates that we can pass directly to the partners API endpoint. They won't lie exactly along strict definitions of what those neighborhoods and cities might have as definitions for borders.
## Editing Places:

places-example
## Editing Cities:

cities-example

### CODE REVIEW

File: assets/javascripts/apps/cities.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        console.error(err);
+        $submit.attr('data-state', 'error');
+      });
+  });
+};
+
+export default {
+  new: () => initEditor(),
+  show: () => initEditor()
+};
```
Comment (dzucconi) [inline assets/javascripts/apps/cities.js:98]: All of this client-side stuff is really rough and has a lot of duplication. Ultimately there's not a lot we have to do here so I don't intend there to be a particularly heavy client-side anyway.

**Comment by dzucconi** [comment]
I'm also forgetting the main reason I started on this: we want to define a loose hierarchy of places. Country/regions that have sub-region/cities (that maybe contain further sub-regions). So there should be a place to structure all this.

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (mzikherman) [inline .babelrc:1]: babel

Comment (dzucconi) [inline .babelrc:1]: Your thoughtful feedback is always welcome here

File: gulpfile.babel.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
+import fs from 'fs';
```
Comment (mzikherman) [inline gulpfile.babel.js:1]: I see you used gulp as the build tool.

File: apps/cities/views/index.jade [hunk 4]
```diff
@@ -0,0 +1,35 @@
+extends ../../../views/layout
+
+block body
+  .page
+    a.avant-garde-button.is-small( href='/cities/new' )
+      | Create new city
+
+    hr
+
+    table.bordered-table
```
Comment (mzikherman) [inline apps/cities/views/index.jade:10]: table

File: apps/cities/views/index.jade [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+              = city.slug
+            td
+              = city.featured
+            td
+              = city.sort_key
+
+            td: a.avant-garde-button.is-dangerous.is-tiny(
+              href='/cities/#{city.id}'
+              data-method='DELETE'
+              data-confirm='Are you certain?'
```
Comment (mzikherman) [inline apps/cities/views/index.jade:34]: For fun we should make this do: 'Hit Cancel in order to delete' just to reverse people and force them to think

Comment (dzucconi) [inline apps/cities/views/index.jade:34]: :+1:

**Comment by anandaroop** [comment]
Looks great @dzucconi !

Having our own version of this data and an admin app totally makes sense if we want the ability to "gerrymander" places/cities to suit our needs.

The mockup seems to show plain JSON, but I'd consider using GeoJSON as the de facto standard for geospatial interchange. Though it is maybe 10-20% more verbose, it could provide many benefits, such as the ability to make use of spatial indexes in Mongo.

Of course if you need this data on the client side, you could optimize and serve a slimmed down version, but client-side libs might prefer GeoJSON too.

As for the hierarchy stuff -- that can get tricky depending on what it's for. But if I'm remembering the issue correctly, this is more a question of presenting nested options in a "Select a city" pulldown than it is a question of modeling containment hierarchies of actual geometries, right? If so, this seems reasonable.

Good start!

**Comment by joeyAghion** [comment]
Re. the hierarchy, we should use existing data sources!

I've used Yahoo's WOEIDs at other companies. There are freely downloadable databases, and the IDs are already hierarchical and can be queried by parent or granularity.

**Comment by alexanderfabry** [comment]
This is very awesome @dzucconi!

**Comment by anandaroop** [comment]
@joeyAghion I mentioned this in https://github.com/artsy/places/issues/3#issuecomment-163312612 but Mapzen has a new data product called Who's On First, which aggregates from Yahoo's GeoPlanet (the source for the WOEIDs) as well as many other canonical sources, such as the Getty's Thesaurus of Geographic Names, Quattroshapes, Geonames, Quattroshapes etc

Currently it's more of a huge data dump of geojson files, available from Github or S3, rather than a web service per se. But it will often include detailed geometries rather than just the centroids and bounding boxes, as Yahoo's does. And it does model containment as well as other goodies like localized name variants.

**Pull request merged**